### PR TITLE
Integrated ODC into dashboard

### PR DESCRIPTION
**Pull request by powerfooI**
## Summary

1. Added `auth` package for generating and validating auth tokens
2. Added a new channel `ODC` for connection creation method

For #371

### CODE REVIEW

File: internal/dashboard/router/v1/info_router.go [hunk 1]
```diff
@@ -21,4 +21,5 @@ import (
 func InitInfoRoutes(g *gin.RouterGroup) {
 	g.GET("/info", h.Wrap(h.GetProcessInfo))
 	g.GET("/statistics", h.Wrap(h.GetStatistics))
+	g.PATCH("/configurable-infos", h.Wrap(h.ConfigureInfo))
```
Comment (chris-sun-star) [inline internal/dashboard/router/v1/info_router.go:24]: patch info directly is ok

File: internal/dashboard/router/v1/user_router.go [hunk 2]
```diff
@@ -21,4 +21,5 @@ import (
 func InitUserRoutes(g *gin.RouterGroup) {
 	g.POST("/login", h.Wrap(h.Login))
 	g.POST("/logout", h.Wrap(h.Logout))
+	g.POST("/auth/:token", h.Wrap(h.Authz))
```
Comment (chris-sun-star) [inline internal/dashboard/router/v1/user_router.go:24]: Why not just use Auth

File: internal/dashboard/handler/helper.go [hunk 3]
```diff
@@ -47,3 +54,53 @@ func loggingCreateOBTenantParam(param *param.CreateOBTenantParam) {
...
+	odcBastionData := OdcBastionData{
+		AccountVerifyToken: token.String(),
+		Type:               "OB_MYSQL",
+		UnionDbUser:        user,
+		Host:               host,
+		Password:           passwd,
+		Port:               2881,
+	}
+
+	odcBastion := OdcBastion{
```
Comment (chris-sun-star) [inline internal/dashboard/handler/helper.go:96]: is this only for unencrypted data

File: internal/dashboard/model/response/connection.go [hunk 4]
```diff
@@ -18,8 +18,10 @@ type OBConnection struct {
 	Tenant     string `json:"tenant,omitempty"`
 	Pod        string `json:"pod"`
 	ClientIP   string `json:"clientIp"`
-	TerminalID string `json:"terminalId"`
+	TerminalID string `json:"terminalId,omitempty"`
 	User       string `json:"user"`
 	Password   string `json:"-"`
 	Host       string `json:"-"`
+
+	OdcVisitURL string `json:"odcVisitURL,omitempty"`
```
Comment (chris-sun-star) [inline internal/dashboard/model/response/connection.go:26]: OdcURL or OdcConnectionURL may be a proper name

File: internal/dashboard/router/v1/user_router.go (hunk 2 above)
Comment (powerfooI) [reply internal/dashboard/router/v1/user_router.go:24]: `Auth` is confusing that could stand for Authentication and Authorization, while `Authz` is the latter.

File: internal/dashboard/handler/helper.go (hunk 3 above)
Comment (powerfooI) [reply internal/dashboard/handler/helper.go:96]: Yes, the encryption algorithm (`SecureRandom` and `KeyGenerator`) was written in Java and there is no corresponding implementation in Go.

Comment (chris-sun-star) [reply internal/dashboard/handler/helper.go:96]: OK

**Review by chris-sun-star — APPROVED** [review]

**Pull request merged**